docs/reference/actions.md

```diff
@@ -65,11 +65,70 @@ input type=submit | click
 select            | change
 textarea          | input
 
+
+
+## KeyboardEvent Filter
+
+You may have Actions that you wish to run only when certain keystrokes are received.
+
+You can install an event listener that responds only to the `Escape` key by adding `.esc` to the event name of the action descriptor, as in the following example.
+
+```html
+<div data-controller="modal"
+     data-action="keydown.esc->modal#close" tabindex="0">
+</div>
+```
+
+This will only work if the event being fired is a keyboard event.
+
+The correspondence between these filter and keys is shown below.
+
+Filter | Key Name
+-------- | --------
+enter    | Enter
+tab      | Tab
+esc      | Escape
+space    | " "
+up       | ArrowUp
+down     | ArrowDown
+left     | ArrowLeft
+right    | ArrowRight
+home     | Home
+end      | End
+[a-z]    | [a-z]
+[0-9]    | [0-9]
+
+If you need to support other keys, you can customize the modifier using custom schema.
+
+```javascript
+import { Application, defaultSchema } from "@hotwired/stimulus"
+
+const customSchema = {
+  ...defaultSchema,
+  keyMappings: {...defaultSchema.keyMappings, "@": "@" },
+}
+
+const app = Application.start(document.documentElement, customSchema);
+```
+
+If you want to subscribe to a compound filter using a modifier key, you can write it like `shift+a`.
+
+```html
+<div data-action="keydown.shift+a->listbox#selectAll" role="option" tabindex="0">...</div>
+```
+
+The list of supported modifier keys is shown below.
```

Comment:
Is this last section needed? This seems to conflict with the table above.

Reply:
@lb- 
I included the list of modifier keys because I thought it should be explicit.
How does this conflict with the table above?

Reply:
sorry, my bad - I was confused, I will suggest a table approach in a min if that is ok.

Reply:
Certainly it would be easier to use a table structure and include explanations, given the existence of meta keys, etc.
Thanks for your kind review.
It is really helpful.